File: python/Maximum Total Importance of Roads.py

def maximumImportance(n: int, roads) -> int:
    degree = [0] * n

    for edge in roads:
        degree[edge[0]] += 1
        degree[edge[1]] += 1

    degree.sort()

    value = 1
    total_importance = 0
    for d in degree:
        total_importance += value * d
        value += 1

    return total_importance
print(maximumImportance( 5,  [[0,1],[1,2],[2,3],[0,2],[1,3],[2,4]]))

File: JAVA/java/src/MaximumTotalImportanceofRoads.java
import java.util.Arrays;

public class MaximumTotalImportanceofRoads {
    public static long maximumImportance(int n, int[][] roads) {
        long[] degree = new long[n];

        for (int[] edge : roads) {
            degree[edge[0]]++;
            degree[edge[1]]++;
        }

        Arrays.sort(degree);

        long value = 1;
        long totalImportance = 0;
        for (long d : degree) {
            totalImportance += (value * d);
            value++;
        }

        return totalImportance;
    }

    public static void main(String[] args) {
        System.out.println(maximumImportance( 5,new int[][]{{0,1},{1,2},{2,3},{0,2},{1,3},{2,4}}));
    }
}

